File: Java/프로그래머스/1/142086. 가장 가까운 같은 글자/가장 가까운 같은 글자.java
class Solution {
    public int[] solution(String s) {
        int[] answer = new int[s.length()];
        int[] alpha = new int[26]; //alphabet using asciicode
        int[] idx = new int[26]; //index saving
        for(int i = 0; i < s.length(); i++){
            char ch = s.charAt(i);
            
            if(alpha[ch-'a'] == 0){
                answer[i] = -1;
                alpha[ch-'a']++;
            }
            else{
                answer[i] = i - idx[ch-'a'];
            }
            idx[ch-'a'] = i;
        }
        
        return answer;
    }
}

File: Python3/프로그래머스/1/142086. 가장 가까운 같은 글자/가장 가까운 같은 글자.py
def solution(s):
    answer = []
    # 쌍으로 하는게 편하지 않을까..?
    dic = {}
    for idx, txt in enumerate(list(s)):
        if txt not in dic:
            answer.append(-1)
            # 단어의 인덱스? 정보? 저장 (b:0,a:1,...처럼 저장되서 키값이 영어, 밸류값을 이용한 계산 만들기)
            dic[txt] = idx
        else:
            answer.append(idx - dic[txt])
            dic[txt] = idx
    return answer